### ISSUE TITLE
Incorrect error requiring :this() initialize in static constructors of positional records make it impossible to declare a static ctor

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#50170]
**Version Used**: 
```
Branch master (27 Dec 2020)
Latest commit cb8f9ae by CyrusNajmabadi:
Merge pull request #50143 from paul1956/CleanUpCA1805

Clean up CA1805 in Visual Basic from approved list of files
```

**Steps to Reproduce**:

Try to compile the following code
```cs
record R(int x)
{
  // error CS8862: A constructor declared in a record with parameter list 
  //               must have 'this' constructor initializer.
  static R() { }
}
```

**Expected Behavior**:
The code compiles

**Actual Behavior**:
`CS8862: A constructor declared in a record with parameter list must have 'this' constructor initializer.` error is reported.

**Notes**
The issue makes it impossible to declare a static constructor in a positional record since static constructors aren't allowed to have `this()` or `base()` initializers
